internal/common/utils/DynamicMockTranslator.go

```diff
@@ -0,0 +1,95 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RedHatInsights/tenant-utils/pkg/tenantid"
+)
+
+type dynamicMockTranslator struct {
+	tenantid.BatchTranslator
```

Comment:
is this needed?

Reply:
It looks like the `tenantid.Translator` interface requires a `BatchTranslator`. I'm not sure of a cleaner way to define it here.

Reply:
It does but what that means is that all the methods from BatchTranslator need to be implemented by whomever implements the Translator interface. It does not require the implementation to be nested like this.